content/docs/io/overview.md

```diff
@@ -6,7 +6,109 @@ menu:
     parent: io
 ---
 
-This page has not been worked on yet. If you'd like to contribute visit the [doc-push]
-repo.
+Rust [`std::io`] module provides traits that synchronously access resources like
+files and sockets.  These `Read` and `Write` traits are implemented in other
+modules including [`std::fs`] and `std::net`.
 
-[doc-push]: https://github.com/tokio-rs/doc-push
+The [`tokio`] crate comes with a variation on those types, [`AsyncRead`] and
+[`AsyncWrite`].  The primary difference between [`AsyncRead`] and `std::io::Read`
+is the [`futures::Async`] return values.  The implementation returns an `Error`,
+`NotReady`, or `Ready` value.
+
+## Non Blocking
+
+The nature of a polling data reader/writer allows for other tasks to operate
+before data reaches the socket or disk queue.  Synchronous operations evaluate
+in a guaranteed sequential manner, and require commands to complete before the
+next operation executes.
+
+Consider this `std::net` implementation from the [`std::net` examples]
+
+```rust,no_run
+# #![allow(unused)]
+# use std::io;
+use std::net::{TcpListener, TcpStream};
+
+# fn handle_client(stream: TcpStream) {
+#     // ...
+# }
+#
+fn main() -> io::Result<()> {
+    let listener = TcpListener::bind("127.0.0.1:8000")?;
+#
+#     // accept connections and process them serially
+    for stream in listener.incoming() {
+        handle_client(stream?);
+    }
+#    Ok(())
+}
+```
+
+This single-threaded process will evaluate through calling `.incoming()` on
+the `TcpListener` and will continue to run until receiving a KILL signal from
+the operating system or until enough data is received on the TCP port to flush
+the buffer, and reevaluating `.incoming()`.
+
+`Tokio` I/O, rather than pausing execution, uses a `Futures` based syntax to
+allow an event loop to respond to I/O rather than wait for it.  For network I/O
```

Comment:
Maybe something like:

> [...] based syntax allowing tasks to respond to I/O events as they arrive rather than wait for them. Networking events are provided by `mio`, a non-blocking I/O library backed by the operating system's event queue (`epoll`, `kqueue`, `iocp`, ...).